src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/vmware/VMWareProvider.java

```diff
@@ -80,15 +84,43 @@
     @Override
     public Set<Instance> createInstance(Infrastructure infrastructure, Instance instance) {
 
+        String instanceImageId;
+        ResourcePool destinationPool = null;
+        HostSystem destinationHost = null;
+        Datastore destinationDatastore = null;
+        Folder destinationFolder = null, finalDestinationFolder;
+
         try {
+            Folder rootFolder = vmWareServiceInstanceCache.getServiceInstance(infrastructure).getRootFolder();
 
-            String instanceFolder = instance.getImage().split("/")[0];
-            String instanceImageId = instance.getImage().split("/")[1];
+            if (instance.getImage().contains("/")) {
+                instanceImageId = instance.getImage().split("/")[0];
+                String hostname = instance.getImage().split("/")[1];
+                if (hostname.equals("*")) {
```

Comment:
You could put the "*" into a class variable. That makes changes easier and you could give it a more descriptive name. I am wondering right now: is it a convention our code or others code or vmware? A variable name could answer those questions.